problem_builder/tasks.py

```diff
@@ -108,4 +106,60 @@ def scan_for_blocks(block):
         "report_filename": filename,
         "start_timestamp": start_timestamp,
         "generation_time_s": generation_time_s,
+        "display_data": [] if len(rows) == 1 else rows[1:]
     }
+
+
+def _extract_data_for_display(submission, student_id, block_type, match_string):
+    """
+    Extract data that will be displayed on Student Answers Dashboard
+    from `submission`.
+    """
+
+    # Username
+    user = user_by_anonymous_id(student_id)
+    username = user.username
+
+    # Question
+    student_item = StudentItem.objects.get(pk=submission['student_item'])
+
+    block_key = UsageKey.from_string(student_item.item_id)
+    block = modulestore().get_item(block_key)
```

Comment:
You should not use the `StudentItem` class since it's essentially a private class within the submissions API. I don't think you actually need any of the three lines above - if you look at where you're calling `_extract_data_for_display()`, you already have the XBlock loaded as `block`, so there is no need to go through the student_item>block_key>block lookup, which is inefficient. Just pass `block` in as a parameter to `_extract_data_for_display`.